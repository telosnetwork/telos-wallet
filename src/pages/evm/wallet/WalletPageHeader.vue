<script setup lang="ts">
import { computed } from 'vue';
import { useRouter } from 'vue-router';
import InlineSvg from 'vue-inline-svg';
import BigFiatText from 'components/evm/BigFiatText.vue';
import { useChainStore } from 'src/antelope';
import EVMChainSettings from 'src/antelope/chains/EVMChainSettings';

const router = useRouter();

const props = defineProps<{
    totalBalance: number
}>();

const buyMoreLink = computed(() => {
    const chainStore = useChainStore();
    return (chainStore.currentChain.settings as EVMChainSettings).getBuyMoreOfTokenLink();
});

const goToRoute = (name: string) => {
    router.push({ name });
};

const goToBuy = () => {
    window.open(buyMoreLink.value, '_blank')?.focus();
};

</script>

<template>
<div>
    <BigFiatText :amount="totalBalance" class="q-mb-xl"/>

    <div class="c-wallet-page-header__links">
        <div
            class="c-wallet-page-header__link"
            tabindex="0"
            role="link"
            :aria-label="$t('evm_wallet.link_to_send_aria')"
            @keypress.space.enter="goToRoute('evm-send')"
            @click="goToRoute('evm-send')"
        >
            <InlineSvg
                :src="require('src/assets/icon--send.svg')"
                aria-hidden="true"
                :alt="$t('evm_wallet.send_icon_alt')"
                class="c-wallet-page-header__icon"
            />
            <span class="c-wallet-page-header__link-text">
                {{ $t('evm_wallet.send') }}
            </span>
        </div>

        <div
            class="c-wallet-page-header__link"
            tabindex="0"
            role="link"
            :aria-label="$t('evm_wallet.link_to_receive_aria')"
            @keypress.space.enter="goToRoute('evm-receive' )"
            @click="goToRoute('evm-receive')"
        >
            <InlineSvg
                :src="require('src/assets/icon--send.svg')"
                aria-hidden="true"
                :alt="$t('evm_wallet.receive_icon_alt')"
                class="c-wallet-page-header__icon c-wallet-page-header__icon--rotated"
            />
            <span class="c-wallet-page-header__link-text">
                {{ $t('evm_wallet.receive') }}
            </span>
        </div>

        <div
            class="c-wallet-page-header__link"
            tabindex="0"
            role="link"
            :aria-label="$t('evm_wallet.link_to_buy_aria')"
            @keydown.space.enter="goToRoute('buy')"
            @click="goToRoute('buy')"
        >
            <InlineSvg
                :src="require('src/assets/icon--buy.svg')"
                aria-hidden="true"
                :alt="$t('evm_wallet.buy_icon_alt')"
                class="c-wallet-page-header__icon"
            />
            <span class="c-wallet-page-header__link-text">
                {{ $t('evm_wallet.buy') }}
            </span>
        </div>
    </div>
</div>
</template>

<style lang="scss">
.c-wallet-page-header {
    $this: &;

    &__links {
        display: flex;
        gap: 48px;
        justify-content: center;
    }

    &__link {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        gap: 8px;
        width: max-content;
        cursor: pointer;
    }

    &__icon {
        transition: transform 0.2s ease;

        &:not(#{$this}__icon--rotated):hover {
            transform: scale(1.05);
        }

        // svg overrides
        path {
            fill: $primary;
        }

        rect {
            stroke: $primary;
        }

        &--rotated {
            transform: rotate(180deg);
            &:hover {
                transform: scale(1.05) rotate(180deg);
            }
        }
    }

    &__link-text {
        @include text--header-4;
        color: var(--text-high-contrast);
        text-align: center;
        width: max-content;
    }
}
</style>
