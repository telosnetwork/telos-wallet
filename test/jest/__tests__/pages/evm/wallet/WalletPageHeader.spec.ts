import { shallowMount } from '@vue/test-utils';

const fakeBuyMoreLink = 'test';
const chainStoreMock = {
    useChainStore: () => ({
        currentChain: {
            settings: {
                getBuyMoreOfTokenLink: () => fakeBuyMoreLink,
            },
        },
    }),
};

import WalletPageHeader from 'pages/evm/wallet/WalletPageHeader.vue';

jest.mock('src/antelope', () => chainStoreMock);

describe('WalletPageHeader.vue', () => {
    const routerMock = {
        push: jest.fn(),
    };

    it('should have the correct name', () => {
        expect(WalletPageHeader.name).toBe('WalletPageHeader');
    });

    it('should render properly', async () => {
        const originalWindowOpen = window.open;

        const wrapper = shallowMount(WalletPageHeader, {
            global: {
                mocks: {
                    $router: routerMock,
                },
            },
        });

        window.open = jest.fn();

        expect(wrapper.element).toMatchSnapshot();

        const [
            sendLink,
            receiveLink,
            buyLink,
        ] = wrapper.findAll('.c-wallet-page-header__link');

        await sendLink.trigger('click');
        await receiveLink.trigger('click');
        await buyLink.trigger('click');

        expect(routerMock.push).toHaveBeenCalledTimes(2);
        expect(routerMock.push.mock.calls[0][0]).toEqual(expect.objectContaining({
            name: 'evm-send',
        }));
        expect(routerMock.push.mock.calls[1][0]).toEqual(expect.objectContaining({
            name: 'evm-receive',
        }));

        expect(window.open).toHaveBeenCalledTimes(1);
        expect(window.open).toHaveBeenCalledWith(fakeBuyMoreLink, '_blank');

        window.open = originalWindowOpen;
    });
});
