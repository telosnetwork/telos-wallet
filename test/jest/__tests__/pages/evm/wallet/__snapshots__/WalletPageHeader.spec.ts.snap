// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletPageHeader.vue should render properly 1`] = `
<div>
  <big-fiat-text-stub
    amount="123456.78"
    class="q-mb-xl"
  />
  <div
    class="c-wallet-page-header__links"
  >
    <div
      aria-label="evm_wallet.link_to_send_aria"
      class="c-wallet-page-header__link"
      role="link"
      tabindex="0"
    >
      <inline-svg-stub
        alt="evm_wallet.send_icon_alt"
        aria-hidden="true"
        class="c-wallet-page-header__icon"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
      <span
        class="c-wallet-page-header__link-text"
      >
        evm_wallet.send
      </span>
    </div>
    <div
      aria-label="evm_wallet.link_to_receive_aria"
      class="c-wallet-page-header__link"
      role="link"
      tabindex="0"
    >
      <inline-svg-stub
        alt="evm_wallet.receive_icon_alt"
        aria-hidden="true"
        class="c-wallet-page-header__icon c-wallet-page-header__icon--rotated"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
      <span
        class="c-wallet-page-header__link-text"
      >
        evm_wallet.receive
      </span>
    </div>
    <div
      aria-label="evm_wallet.link_to_buy_aria"
      class="c-wallet-page-header__link"
      role="link"
      tabindex="0"
    >
      <inline-svg-stub
        alt="evm_wallet.buy_icon_alt"
        aria-hidden="true"
        class="c-wallet-page-header__icon"
        keepduringloading="true"
        src=""
        transformsource="[Function]"
      />
      <span
        class="c-wallet-page-header__link-text"
      >
        evm_wallet.buy
      </span>
    </div>
  </div>
</div>
`;
